e2e/nomostest/gitproviders/git-provider.go

```diff
@@ -63,3 +63,17 @@ func NewGitProvider(t testing.NTB, provider string, logger *testlogger.TestLogge
 		return &LocalProvider{}
 	}
 }
+
+// WithLogger updates an existing GitProvider to use the provider TestLogger
+func WithLogger(t testing.NTB, provider GitProvider, logger *testlogger.TestLogger) GitProvider {
+	switch p := provider.(type) {
+	case *BitbucketClient:
+		prevLogger := p.logger
+		t.Cleanup(func() {
```

Comment:
why revert the logger on cleanup?

Reply:
so that it uses the original logger after the test is finished, e.g. in the case of a shared test env if there's any cleanup that's done at the end of the test suite